src/sentry/rules/conditions/event_attribute.py

```diff
@@ -1,38 +1,41 @@
-from typing import Any, Sequence
+from typing import Any, Dict, Sequence
 
 from django import forms
 
 from sentry.eventstore.models import GroupEvent
 from sentry.rules import MATCH_CHOICES, EventState, MatchType
 from sentry.rules.conditions.base import EventCondition
-
-ATTR_CHOICES = [
-    "message",
-    "platform",
-    "environment",
-    "type",
-    "error.handled",
-    "error.unhandled",
-    "exception.type",
-    "exception.value",
-    "user.id",
-    "user.email",
-    "user.username",
-    "user.ip_address",
-    "http.method",
-    "http.url",
-    "http.status_code",
-    "sdk.name",
-    "stacktrace.code",
-    "stacktrace.module",
-    "stacktrace.filename",
-    "stacktrace.abs_path",
-    "stacktrace.package",
-]
+from sentry.snuba.events import Columns
+from sentry.types.condition_activity import ConditionActivity
+
+# Maps attributes to snuba columns
+ATTR_CHOICES = {
+    "message": Columns.MESSAGE,
+    "platform": Columns.PLATFORM,
+    "environment": Columns.MESSAGE,
+    "type": Columns.TYPE,
+    "error.handled": Columns.ERROR_HANDLED,
+    "error.unhandled": Columns.ERROR_HANDLED,
+    "exception.type": Columns.ERROR_TYPE,
+    "exception.value": Columns.ERROR_VALUE,
+    "user.id": Columns.USER_ID,
+    "user.email": Columns.USER_EMAIL,
+    "user.username": Columns.USER_USERNAME,
+    "user.ip_address": Columns.USER_IP_ADDRESS,
+    "http.method": Columns.HTTP_METHOD,
+    "http.url": Columns.HTTP_URL,
+    "http.status_code": Columns.HTTP_STATUS_CODE,
+    "sdk.name": Columns.SDK_NAME,
+    "stacktrace.code": None,
```

Comment:
I can't find the corresponding column for this attribute, the stacktrace code should be made up of `pre_context`, `context`, and `post_context`, do you know if these are queryable?

Reply:
special case: we are looking for the `frame` `context_line`. see `test_stacktrace_code` in `tests/sentry/rules/conditions/test_event_attribute.py`

Reply:
decided that we're not doing this at least for now